### PR TITLE
repo: remove low-hanging deprecations

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

The pull-request includes multiple commits addressing the removal of deprecations present in the framework.
The methods, interfaces, constants and re-exports were deprecated for some time, and a cleanup was in order.

#### How to test

Confirm that CI passes (both builds and tests).

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: packages/preferences/src/browser/util/preference-types.ts [hunk 1]
```diff
@@ -21,10 +21,10 @@ import {
     CompositeTreeNode as BaseCompositeTreeNode,
     PreferenceInspection,
     CommonCommands,
-    JsonType,
 } from '@theia/core/lib/browser';
 import { Command, MenuPath } from '@theia/core';
 import { JSONValue } from '@theia/core/shared/@phosphor/coreutils';
+import { JsonType } from '@theia/core/src/common/json-schema';
```
Comment (colin-grant-work) [inline packages/preferences/src/browser/util/preference-types.ts:27]: Lint

File: packages/core/src/browser/context-menu-renderer.ts [hunk 2]
```diff
@@ -79,12 +79,8 @@ export abstract class ContextMenuRenderer {
         }
     }
 
-    render(options: RenderContextMenuOptions): ContextMenuAccess;
-    /** @deprecated since 0.7.2 pass `RenderContextMenuOptions` instead */
-    render(menuPath: MenuPath, anchor: Anchor, onHide?: () => void): ContextMenuAccess;
-    render(menuPathOrOptions: MenuPath | RenderContextMenuOptions, anchor?: Anchor, onHide?: () => void): ContextMenuAccess {
-        const resolvedOptions = RenderContextMenuOptions.resolve(menuPathOrOptions, anchor, onHide);
```
Comment (colin-grant-work) [inline packages/core/src/browser/context-menu-renderer.ts:86]: This function used to be responsible for adding (or not adding) the `anchor` arg to the args passed in - at least that logic needs to be carried over, I think.

**Review by colin-grant-work — COMMENTED** [review]
All looks to be in order to me. Only one comment.

**Review by colin-grant-work — APPROVED** [review]
This looks good to me. The code is cleaner, the tests pass, and the changelog entry appears comprehensive.

**Comment by vince-fugnitto** [comment]
I'll merge tomorrow if there are no objections.

**Pull request merged**